identity-service/src/routes/solana.js

```diff
@@ -19,14 +19,20 @@ const {
 const solanaRouter = express.Router()
 const connection = new Connection(solanaEndpoint)
 
+// Check that an instruction has all the necessary data
+const isValidInstruction = (instr) => {
+  if (!instr || !Array.isArray(instr.keys) || !instr.programId || !instr.data) return false
+  if (!instr.keys.every(key => !!key.pubkey)) return false
+  return true
+}
+
 solanaRouter.post('/relay', handleResponse(async (req, res, next) => {
   const redis = req.app.get('redis')
-  const { recentBlockhash, secpInstruction, instruction } = req.body
+  const { recentBlockhash, secpInstruction, instruction = {}, instructions = [] } = req.body
 
   const reqBodySHA = crypto.createHash('sha256').update(JSON.stringify({ secpInstruction, instruction })).digest('hex')
 
   try {
-    if (!instruction) return errorResponseBadRequest('Please provide transaction instruction')
     const tx = new Transaction({ recentBlockhash })
 
     if (secpInstruction) {
```

Comment:
The new libs code will call with multiple secp instructions per transaction - does it make sense to still have this? Is this just for backwards compat?

Reply:
I left it for backwards compat. If you look at the libs changes, it just encodes the secp instr as a regular instruction.